app/components/dashboard/dashboard.controller.js

```diff
@@ -0,0 +1,173 @@
+(function () {
+
+    'use strict';
+
+    angular.module('FoodOrderingApp')
+        .controller('DashboardController', DashboardController);
+
+    DashboardController.$inject = ['$state', '$uibModal', '$sessionStorage', '$log', 'APP_CONSTANT', 'RestaurantService', 'OrderService'];
+
+    function DashboardController($state, $uibModal, $sessionStorage, $log, APP_CONSTANT, RestaurantService, OrderService) {
+
+        var vm = this;
+
+        vm.noFoodMsg = APP_CONSTANT.NO_FOOD_MSG;
+        vm.restaurants = [];
+        vm.message = '';
+        vm.firstName = $sessionStorage.firstName;
+        vm.role = $sessionStorage.role;
+
+        vm.initRestaurants = function () {
+            RestaurantService.getRestaurantList().then(
+                function (answer) {
+                    vm.restaurants = answer.data;
+                },
+                function (error) {
+                    $log.info(error);
+                }
+            );
+        };
+        vm.initRestaurants();
```

Comment:
Use `vm.$onInit` instead. Something like this:
```
vm.$onInit = function () {
            RestaurantService.getRestaurantList().then(
                function (answer) {
                    vm.restaurants = answer.data;
                },
                function (error) {
                    $log.info(error);
                }
            );
        };
```
Then you don't need `vm.initRestaurants();`

Reply:
Also make sure to add test spec to check if `RestaurantService.getRestaurantList` has been called and if `vm.restaurants` is set on success and `$log.info` is being called on error.